src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -1033,10 +1033,7 @@ private MethodSymbol PerformForEachPatternOverloadResolution(TypeSymbol patternT
             }
 
             overloadResolutionResult.Free();
-            if (arguments.Arguments.Count == 0)
-            {
-                arguments.Free();
-            }
+            arguments.Free();
```

Comment:
In reviewing the merge from feature branch to dev16, I spotted this (harmless) bug.
It was introduced when `arguments` became a local instead of an optional parameter (https://github.com/dotnet/roslyn/pull/30280/files#diff-fdd3e98ae3d8ab6c5d9f5f274dbee9d7R983)